商品出品機能の実装

tomo1107 opened the pull request:
# What
## 実装内容
- Item(商品)テーブルの実装
- 商品出品機能の実装
- モデルの単体テストの実装
- ログイン状態でのみ商品出品ページへ遷移出来ること

## 動作(Gyazo画像)
- ログイン状態のユーザーは、商品出品ページへ遷移できる動画
https://gyazo.com/c673149097e345fca3fecdb845ccd63f
- 金額が入力されると同時に、販売手数料と販売利益が表示される動画
https://gyazo.com/2f82f28382c5cde9d079cae824c5643a
- 正しく情報を記入すると、商品を出品できる動画
https://gyazo.com/ee40c55b8ce40a56ceb015c57e77cf1d
- 入力に問題のある状態では商品は出品できず、エラーメッセージが出力される動画
https://gyazo.com/9fb5b91b7e584270f4ee3e9e8048328f
- ログアウト状態のユーザーは、商品出品ページへ遷移しようとすると、ログインページへ遷移する動画
https://gyazo.com/49d674418d0c79a54abd7ec446d0a504
- テスト結果の画像
https://gyazo.com/6031020ba86298a6ede967ea04fdae39

# Why
- 商品出品機能が正常に実装出来ているかを確認頂きたい。
- モデルの単体テストが正しく実装されているかを確認頂きたい。

tomo1107 committed:
itemテーブルの作成とバリデーションの設定

tomo1107 committed:
商品出品機能の実装

tomo1107 committed:
商品出品機能のモデルの単体テストの実装

chobi555 commented on app/controllers/items_controller.rb:
```diff
@@ -1,4 +1,26 @@
 class ItemsController < ApplicationController
+  before_action :authenticate_user!, only: [:new, :edit, :destroy]
```
`before_action :authenticate_user!, only: [:new, :edit, :destroy]`
不正アクセス防止のため、上記にcreateも適用させておきましょう。
また、edit, destroyは対象の実装まで消しておきましょう！

tomo1107 replied:
ご指摘の通り、是正しました。

chobi555 commented on spec/models/item_spec.rb:
```diff
@@ -0,0 +1,121 @@
+        @item.valid?
+        expect(@item.errors.full_messages).to include("Description can't be blank")
+      end
+      it '値段が空白では出品できないこと' do
```
priceのテスト項目に下記内容を追記しておきましょう！

```
・半角英数混合では登録できないこと
・全角文字では登録できないこと
・半角英語だけでは登録できないこと
```

tomo1107 replied:
ご指摘の通り、是正しました。

chobi555 reviewed (commented):
レビュー依頼ありがとうございます！
コメント致しました。
確認後、修正をお願い致します。

tomo1107 committed:
レビュー後の是正対応1回目

以下のご指摘に対して、是正を行いました。
https://github.com/tomo1107/furima-35073/pull/3#pullrequestreview-634402652

tomo1107 commented:
> レビュー依頼ありがとうございます！
> コメント致しました。
> 確認後、修正をお願い致します。

ご確認ありがとうございました！
ご指摘を修正しましたので、再レビューの程、よろしくお願いします！

以下はモデルの単体テストの実行結果の画像です。
https://gyazo.com/ea8bab8f69becc6492ed025a9df7d5de

Hide-Morishita commented:
レビュー依頼ありがとうございます。
LGTMとさせていただきます。実装お疲れ様でした。

マージ後、ローカルのmasterブランチに忘れずにプルしましょう。
その後、新しいブランチを作成し次の実装に取り組んでいただけますと幸いです。
引き続き実装頑張っていきましょう。
